sdk-extensions/incubator/src/main/java/io/opentelemetry/sdk/extension/incubator/fileconfig/LoggerProviderFactory.java

```diff
@@ -97,10 +100,78 @@ private static class LoggerConfigFactory
     @Override
     public LoggerConfig create(
         ExperimentalLoggerConfigModel model, DeclarativeConfigContext context) {
+      LoggerConfigBuilder configBuilder = LoggerConfig.builder();
       if (model.getDisabled() != null && model.getDisabled()) {
-        return LoggerConfig.disabled();
+        configBuilder.setEnabled(false);
       }
-      return LoggerConfig.defaultConfig();
+      if (model.getTraceBased() != null && model.getTraceBased()) {
+        configBuilder.setTraceBased(true);
+      }
+      if (model.getMinimumSeverity() != null) {
+        configBuilder.setMinimumSeverity(
+            SeverityNumberFactory.INSTANCE.create(model.getMinimumSeverity(), context));
+      }
+      return configBuilder.build();
+    }
+  }
+
+  // Visible for testing
+  static class SeverityNumberFactory implements Factory<SeverityNumber, Severity> {
```

Comment:
does this need to be a factory? Could it just be a static method?

Reply:
Oh yes much better